specification/common/attribute-precedence.md

```diff
@@ -0,0 +1,199 @@
+attributes are considered to be at the top of the hierarchy, since they are the
+most general attributes. 
+Attributes on individual Spans/ Metric data points/Logs are at the bottom of the
+hierarchy, as they are most specialized and only apply to a subset of all data.
+
+**A more specialized attribute that shares an attribute key with more general 
+attribute will take precedence.** 
+
+Therefore, more specialized attributes will overwrite more general ones.
+In some cases it might be desirable to overwrite an attribute like this.
+<!-- TODO example -->
+
+When de-normalizing the structured OTLP message to a flat set of key-value 
+pairs, attributes that are present at Resource and InstrumentationLibraryScope 
+level will be duplicated for each Span/Metric data point/Log.
+
+## Precedence per Signal
+
+Below, the precedence for each of the signals is spelled out explicitly.
+Some attributes (e.g., Span Link attributes) should only be used when flattening
+out attributes to transform the respective concept (Span Link attributes should
+not be added when transforming attributes on spans, for example).
+
+`A > B` denotes that the attribute on `A` will overwrite the attribute on  `B`
+if the keys clash.
+
+### Traces
+
+```
+Span.attributes > ScopeSpans.scope.attributes > ResourceSpans.resource.attributes
+```
+
+#### Span Events
+
+```
+Span.events.attributes > Span.attributes > ScopeSpans.scope.attributes > ResourceSpans.resource.attributes
+```
+
+#### Span links
+
+```
+Span.links.attributes > Span.attributes > ScopeSpans.scope.attributes > ResourceSpans.resource.attributes
+```
+
+### Metrics
+
+Metrics are different from Spans and LogRecords, as each Metric has a data field
+which can contain one or more data points.
+Each data point has a set of attributes, which need to be considered 
+independently.
+
+```
+Metric.data.data_points.attributes > ScopeMetrics.scope.attributes > ResourceMetrics.resource.attributes
+```
+
+#### Metric exemplars
+
+```
+Metric.data.data_points.exemplars.filtered_attributes > Metric.data.data_points.attributes > ScopeMetrics.scope.attributes > ResourceMetrics.resource.attributes
+```
+
+### Logs
+
+```
+LogRecord.log_records.attributes > ScopeLogs.scope.attributes > ResourceLogs.resource.attributes
+```
+
+## Considerations
+
+Note that this precedence is a strong suggestion, not a requirement.
+Code that transforms attributes should follow this mode of flattening, but might 
+diverge if they have a reason to do so. 
+Furthermore, exporters can apply clash prevention, e.g., by prefixing all 
+Resource attributes with `resource.`.
+Note that even then, a Span/Metric data point/LogRecord attribute can overwrite
+the resource attribute `attribute_name`, if it is called 
+`resource.attribute_name`.
+Therefore, extra care needs to be taken when prefixing attributes.
+
+## Example
+
+De-duplication can be thought of as a map with unique keys to which the 
```

Comment:
Hum.. I'm not sure if we need to explain all this again. The content before makes it pretty clear the order of how things work and how one overrides the other. 

Maybe you can just remove all of this and start with "The following YAML is a representation of an OTLP message.."